Web/src/components/room/hooks/useErrorableChannel.ts

```diff
@@ -41,13 +43,19 @@ export default function useErrorableChannel(channelId: string): ErrorableChannel
 
       newChannel
         .join()
-        .receive(ChannelResponse.OK, () => setChannel(newChannel))
+        .receive(ChannelResponse.OK, (payload: T) => {
+          setJoinPayload(payload);
+          setChannel(oldChannel => {
+            if (oldChannel != null) oldChannel.leave();
+            return newChannel;
+          });
+        })
         .receive(ChannelResponse.ERROR, reasons => setError([ErrorCause.Other, reasons]))
         .receive(ChannelResponse.TIMEOUT, () => setError([ErrorCause.Timeout, null]));
 
       return currentSocket;
     });
```

Comment:
is there some channel cleanup stuff here we need to do when we stop using this channel? e.g. `channel.disconnect()`

Reply:
Yes, it's on line 49: `oldChannel.leave()`

Reply:
Yes, it's on line 49: `oldChannel.leave()`

Reply:
Ah, ok.